### ISSUE TITLE
For loop and LayoutBuilder [Bug or Wrong Coding]

### ISSUE THREAD

**Issue by maheshkumar2150** [flutter/flutter#44820]
```
List<Widget> _temp = [];    

for(int i = 0; i < 10; i++) {
   if(i == 0) { string = "Hello"; } else { string = "World"; }
   _temp.add(Container(
      child: LayoutBuilder(
        builder: (context, constraint) { 
          return Text(string); //This return line makes problem with the for loop
        })
   ));
}

return _temp;
```

_temp has list of widget which is sent back to Column.

The child in the column should print **Hello** and the remaining children should print **World**

But the return statement in LayoutBuilder is causing some problem.
Because of that, _temp contains ten Text widget with the string **World**. There is no **Hello** in the output

Is it a bug or am i making a mistake?
What is the real solution for this?

**Comment by iapicca** [comment]
Hi @maheshkumar2150 
This platform is not met for assistance on personal code.
Please see https://flutter.dev/community for resources for asking questions like this,
you may also get some help if you post it on Stack Overflow.

Don't forget to declare the variable `string `
```
  String string;
   if(i == 0) { string = "Hello"; } else { string = "World"; }
```

also this approach might make your life easier 

<details>
  <summary>Sample code</summary>

```
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context)=> MaterialApp(home: HomePage());
}

class HomePage extends StatelessWidget {

  @override
  Widget build(BuildContext context) => Scaffold(
    body: ListView(
      children: List<Widget>.generate(10, (i)=> Text(
        i<1
        ?'hello'
        :'world'),),
  ),);
}
```

</details>

Closing, as this isn't an issue with Flutter itself,
if you disagree please write in the comments and I will reopen it
Thank you

**Comment by maheshkumar2150** [comment]
The sample code works perfectly. Only it causes problem when i use **any builder widget (ex. LayoutBuilder)**. Because builder widget has to **return another widget**.

The return statement in that builder function causes the problem.

I made the code to work in two different ways. I just posted here to get some idea about the problem.

```
_temp.add(Container(
      child: Text(string); //this works without any problem.
   ));

```

honestly, Just to learn, i posted the question here. Because i was not able to visualize exactly what is happening.

Thanks
